astropy/units/function/mixin.py

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+from .. import core
+
+class FunctionMixin(object):
+    """Mixin class that makes UnitBase subclasses callable.
+
+    Provides a __call__ method that passes on arguments to a FunctionUnit.
+
+    See units.py and logarithmic.py for usage.
+    """
+    _function_unit_class = None
+
+    def __call__(self, unit=None):
+        if self._function_unit_class is None:
+            raise TypeError("Cannot call unit since corresponding "
+                            "function unit class is not defined.")
+        return self._function_unit_class(physical_unit=unit,
+                                         function_unit=self)
+
+
+class IrreducibleUnit(FunctionMixin, core.IrreducibleUnit):
```

Comment:
I think reusing the same name here is a bit confusing (when used in the `function/units.py` file below).  Maybe call it `IrreducibleFunctionUnit`. Same for `FunctionUnit` below...